### ISSUE TITLE
Websites: where do we go from here?

### ISSUE THREAD

**Issue by thibautjombart** [thibautjombart/adegenet#66]
The old adegenet website is getting a bit outdated:
http://adegenet.r-forge.r-project.org/

The project is now a lot more the product of a community than my own work. A devel page makes sense, and I'm starting to use the wiki here. I think eventually the old adegenet website needs to go, or at least be relocated.

Questions:
- Shall we have 2 separate websites for the devel and stable version?
- If so, shall we move the old website to github pages?

Thoughts anyone?

**Comment by romunov** [comment]
I'm a big fan of the KISS principle. I think a github page (README) with devel and stable sections would be sufficient.

**Comment by thibautjombart** [comment]
I see your point and agree when it comes to finding information. From a
communication point of view though, I am afraid the rather raw project page
on github will put some people off.

**Comment by KlausVigo** [comment]
Just for inspiration. Hadley Wickham just twittered this:
If there was a prize for best #rstats package website, I'm sure http://rich-iannone.github.io/DiagrammeR/  would win it

**Comment by thibautjombart** [comment]
Nice! Looks indeed fancier.. Any idea how it's done?

**Comment by KlausVigo** [comment]
If I would know my phangorn page would look like it. Seems they use a lot of bootstrap.

**Comment by psolymos** [comment]
I've just spent some time on exploring github pages + jekyll + bootstrap/bootswatch/jQuery + MathJax + highlighter + remark.js and the results is this:
http://datacloning.org/
Feel free to fork it, I can help and explain, there are many undocumented features and all this is in a bit of a flux:
http://datacloning.org/

A one-paged theme (similar to that of DiagrammeR's) was used for my personal website: http://peter.solymos.org
https://github.com/psolymos/psolymos.github.io

**Comment by thibautjombart** [comment]
This is waaaay cooler than the current adegenet website :) Will definitely check it out - thanks!

**Comment by KlausVigo** [comment]
That looks great, I think phangorn needs something like this too.

**Comment by psolymos** [comment]
You can change the look & feel in the _config.yml by changing these lines:
```
# Bootswatch theme
theme: flatly
# fancy fonts on/off
fancyfonts: true
# navbar style: inverse, transparent
navbar: inverse
```
See all the themes here: http://bootswatch.com/

**Comment by thibautjombart** [comment]
This is really cool. How long did it take you to master the beast?

**Comment by psolymos** [comment]
It kind of fall into my lap because of a website templating project (http://species.abmi.ca/) where the premise was: we want content for 2000 species up the web and don't care about design. But everyone kept telling me: good, good, but could be nicer. I recall the first streak taking roughly 2 days. A good place to start is the jekyll site: http://jekyllrb.com/ .

**Comment by psolymos** [comment]
It fell into my lap because of a website templating project (http://species.abmi.ca/) where the premise was: we want content for 2000 species up the web and don't care about design. But everyone kept telling me: good, good, but could be nicer. I recall the first streak taking roughly 2 days. A good place to start is the jekyll site: http://jekyllrb.com/ .

**Comment by romunov** [comment]
I thought http://pillow.readthedocs.org/ was nice along side the fact that it resides on github and appears to be rendered from there.

**Comment by thibautjombart** [comment]
Yes, this is nice too! I must prepare the next adegenet teaching week, but will investigate soon.

**Comment by romunov** [comment]
If we want to go something like "documentation website", docster would be a nice choice. http://jekyllthemes.org/themes/docster/

**Comment by thibautjombart** [comment]
Looks nice! I have admittedly little time to look into changes for a website at the moment, but good to have this on the issue ;)

**Comment by romunov** [comment]
Once we settle on what should go on the website, I can try and get this in motion. The only thing that isn't yet clear to me is how to have documentation generated from .Rd files or even roxygen2 documented .R files. I'll ask around.

**Comment by thibautjombart** [comment]
If we want to go down that road, we'd have to generate pages from the Rd
files. adegenet only partially uses roxygen2 (the pain of reforming an old
package...).

Thanks for the offer, that sounds great. I'll try and have a look at
options so far today.



--
Dr Thibaut Jombart
Lecturer, Department of Infectious Disease Epidemiology
Imperial College London
https://sites.google.com/site/thibautjombart/
https://github.com/thibautjombart
Twitter: @TeebzR <https://twitter.com/TeebzR>

**Comment by thibautjombart** [comment]
I remember having headaches trying to get jekyll to work on my linux system (ruby deps etc). So I'd be keen on not relying on it.

So I think so far, my favourites are:
- http://species.abmi.ca/ by @psolymos 
- http://thibautjombart.github.io/adegenet/ 

I had forgotten about the latter, but that may be the easiest course of action as it is generated by github.

I think this one 
http://rich-iannone.github.io/DiagrammeR/ 
looks very fancy but not that good in terms of accessing the info easily. 

What do you guys think?

**Comment by romunov** [comment]
While looking at some docs of docs.ggplot2.org I clicked on the "Built by staticdocs..." at the bottom which links to [Hadley's repo](https://github.com/hadley/staticdocs) that *drumroll* generates html pages from Rd files. Add a fancy bootstrap theme et voila.

**Comment by thibautjombart** [comment]
Yes, but this is not exactly what I want. This would be a website replicating Rd files content. What I am looking for is more general - a proper website (i.e. content of the current website). In fact, I'm looking much more for a revamp of the old website than for new technical content (though it is welcome). New content, yes, but probably not so much technical e.g. a 'contributors' section with pics + short description of the current devel team, etc.

**Comment by romunov** [comment]
I've [clicked through the demos](https://github.com/jekyll/jekyll/wiki/Themes) over the weekend and some of the themes are lovely to say the least. The current github website you set up is only missing a few tabs and it's good to go to be filled with contents. And now I'll shut up and give time to other to express their opinion. :)

**Comment by zkamvar** [comment]
This might be of interest: https://github.com/hrbrmstr/markdowntemplates

**Comment by plantarum** [comment]
Hi there,

I understand updating the website is a big project, and probably not a priority for anyone. One small change I would suggest is to remove the circular referencing between github and R-forge. That is, the first line on the R-forge page (http://adegenet.r-forge.r-project.org/) currently directs users to the github wiki. The first link in the github wiki directs users to the R-forge website. I think at this point the wiki is more up-to-date, so perhaps the link back to R-forge could be removed to clarify where we ought to look for info?

Best,

Tyler
